vyper/semantics/types/function.py

```diff
@@ -469,7 +469,13 @@ def fetch_call_return(self, node: vy_ast.Call) -> Optional[BaseTypeDefinition]:
                 if not isinstance(kwarg.value, vy_ast.NameConstant):
                     raise InvalidType("skip_contract_check must be literal bool", kwarg.value)
             else:
-                validate_expected_type(kwarg.arg, kwarg.value)
+                raise ArgumentException(
+                    (
+                        "Usage of kwarg in Vyper is restricted to gas=, "
```

Comment:
this error message is a big improvement, can we make it even better with a suggestion? without complicating the code too much, would be good to see something like
```
vyper.exceptions.ArgumentException: Usage of kwarg in Vyper is restricted to gas=, value= and skip_contract_check=. Try removing the kwarg: Foo(msg.sender).foo(1, 2)
  contract "tmp/foo.vy", function "foo", line 9:28 
       8 def foo():
  ---> 9     Foo(msg.sender).foo(1, y=2)
  -----------------------------------^
      10

```